From ae19d3dfa0a05937281bfdfb85b221fc945e98d6 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 3 Jun 2020 23:19:01 +0530
Subject: [PATCH] Toggle interface state after proxt settings change

Toggle interface state after proxt settings change to
for proxy to take effect.

Change-Id: I1cdbfcf3c1632a77dbc67c5df5466fa96e8176f3
Tracked-On: OAM-91282
Signed-off-by: Raveendra Babu Chennakesavulu
                        <raveendra.babu.chennakesavulu@intel.com>
---
 java/com/android/server/ethernet/EthernetNetworkFactory.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/java/com/android/server/ethernet/EthernetNetworkFactory.java b/java/com/android/server/ethernet/EthernetNetworkFactory.java
index de0a691..1171b3c 100644
--- a/java/com/android/server/ethernet/EthernetNetworkFactory.java
+++ b/java/com/android/server/ethernet/EthernetNetworkFactory.java
@@ -269,6 +269,8 @@ public class EthernetNetworkFactory extends NetworkFactory {
         void setIpConfig(IpConfiguration ipConfig) {
 
             this.mIpConfig = ipConfig;
+            updateLinkState(false);
+            updateLinkState(true);
         }
 
         boolean statisified(NetworkCapabilities requestedCapabilities) {
-- 
2.17.1

